src/main/kotlin/com/ampnet/walletservice/grpc/blockchain/BlockchainServiceImpl.kt

```diff
@@ -436,11 +424,40 @@ class BlockchainServiceImpl(
     private fun serviceWithTimeout() = serviceBlockingStub
         .withDeadlineAfter(applicationProperties.grpc.blockchainServiceTimeout, TimeUnit.MILLISECONDS)
 
+    private fun postTransactionWithRetries(transaction: String, retryCount: Int): String {
+        logger.info { "Posting transaction (#$retryCount)" }
+        if (retryCount > applicationProperties.grpc.blockchainServiceMaxRetries) {
+            logger.warn { "Retry posting transaction exceeded" }
+            throw GrpcException(ErrorCode.INT_GRPC_BLOCKCHAIN, "Retry exceeded")
+        }
+        try {
+            val response = serviceWithTimeout()
+                .postTransaction(
+                    PostTxRequest.newBuilder()
+                        .setData(transaction)
+                        .build()
+                )
+            logger.info { "Successfully posted transaction: ${response.txHash}" }
+            return response.txHash
+        } catch (ex: StatusRuntimeException) {
+            getErrorDescriptionFromExceptionStatus(ex)?.let { grpcErrorCode ->
+                // throw exception for known errors
+                val errorCode = ErrorCode.INT_GRPC_BLOCKCHAIN
+                errorCode.specificCode = grpcErrorCode.code
+                errorCode.message = grpcErrorCode.message
+                throw GrpcException(errorCode, "Couldn't post transaction")
+            }
+            // retry posting transaction for unknown errors
```

Comment:
In case method getErrorDescriptionFromExceptionStatus returns null we won't have any info what happened. Put some logging right above sleep()....